rmw_fastrtps_shared_cpp/CMakeLists.txt

```diff
@@ -25,12 +25,17 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wthread-safety -Werror)
```

Comment:
> We should not use -Werror just in this package, only here

Agreed - emoving `-Werror`

> should -Wthread-safety always be set or just conditionally when TSan is being enabled?

This analysis is unrelated to TSan, it's this whole other thing https://clang.llvm.org/docs/ThreadSafetyAnalysis.html (that is only actually happening in the nightly `clang-libcxx` build, because the `std::mutex` type is not annotated in gnu's `libstdc++`)

It makes the most sense to me to have it always enabled, it's a compile-time analysis that doesn't happen if the compiler doesn't have the capability

Reply:
Sounds good.